Kevin-Muthiani/Arrays_and_Strings/remove_element.py

```diff
@@ -0,0 +1,25 @@
+# https://leetcode.com/problems/remove-element/
+# Given an array nums and a value val, remove all instances of that value 
+# in-place and return the new length. Do not allocate extra space for another 
+# array, you must do this by modifying the input array in-place with O(1) 
+# extra memory. The order of elements can be changed. It doesn't matter what 
+# you leave beyond the new length.
+
+def removeElement(nums, val):
+    n = 0
+
+    for i in range(len(nums)):
+        if nums[i] != val:
+            nums[n] = nums[i]
+            n += 1
```

Comment:
This correctly gives you the size but it does not remove all the values from the `nums` arr

Reply:
For example, try feeding it an input of nums = [3, 2, 2, 3, 0, 1, 9, 0] and val = 0

Reply:
> For example, try feeding it an input of nums = [3, 2, 2, 3, 0, 1, 9, 0] and val = 0

I agree. My understanding of the question (commented out at the top of the file) was that for such an input,  
nums = [3, 2, 2, 3, 1, 9, X, X] and n = 6  was a valid result. With that in mind, i aimed for a linear time complexity.
For this particular input, the result is: nums = [3, 2, 2, 3, 1, 9, 9, 0] and n = 6